### PR TITLE
Search engine updates

### PR DESCRIPTION
**Pull request by bsclifton**
Adds Ecosia to the following markets:
USA (US), UK (GB), France (FR), Germany (DE), Netherlands (NL), Belgium (BE), Switzerland (CH), Sweden (SE)

Fixes https://github.com/brave/brave-browser/issues/13283

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
TBD

### CODE REVIEW

**Comment by bsclifton** [comment]
@deeppandya anything needed on Android to add this new option to the list?

**Comment by deeppandya** [comment]
> @deeppandya anything needed on Android to add this new option to the list?

@bsclifton we would need new icons for ecosia for search engine onboarding in android. @jamesmudgett or @karenkliu can you help with it ?

File: components/search_engines/brave_prepopulated_engines.cc [hunk 1]
```diff
@@ -152,6 +153,16 @@ const PrepopulatedEngine duckduckgo_lite = {
 };
 #endif
 
+const PrepopulatedEngine brave_ecosia = 
+    ModifyEngineParams(ecosia,
+                       NULL,
+                       L":e",
+                       "https://www.ecosia.org/search?tt="
+                       "42b8ae98"
```
Comment (diracdeltas) [inline components/search_engines/brave_prepopulated_engines.cc:161]: could you start a wiki page that lists all the search refcodes that we use? makes it easier for support to tell users that the ID is non-unique

Comment (bsclifton) [reply components/search_engines/brave_prepopulated_engines.cc:161]: Great idea! Will do 😄

Comment (bsclifton) [reply components/search_engines/brave_prepopulated_engines.cc:161]: @diracdeltas created https://github.com/brave/brave-browser/wiki/Search-engine-integrations 👍

**Review by simonhong — APPROVED** [review]
++ Works great!

**Comment by bsclifton** [comment]
Couple of notes:
- need to update desktop w/ referral ID
- need to update P3A to be able to track this (forgot)
- Android implementation needs icon and work for onboarding screen

File: browser/search_engines/search_engine_tracker.cc [hunk 2]
```diff
@@ -27,10 +27,10 @@ void RecordSearchEngineP3A(const GURL& search_engine_url,
     answer = SearchEngineP3A::kBing;
   } else if (type == SEARCH_ENGINE_QWANT) {
     answer = SearchEngineP3A::kQwant;
-  } else if (type == SEARCH_ENGINE_YAHOO) {
-    answer = SearchEngineP3A::kYahoo;
   } else if (type == SEARCH_ENGINE_YANDEX) {
     answer = SearchEngineP3A::kYandex;
+  } else if (type == SEARCH_ENGINE_ECOSIA) {
```
Comment (iefremov) [inline browser/search_engines/search_engine_tracker.cc:32]: I guess we should also bump the metric id version?

**Review by iefremov — APPROVED** [review]

**Comment by iefremov** [comment]
will need updating https://github.com/brave/brave-browser/wiki/P3A

**Comment by bsclifton** [comment]
@pes10k can you please check out the latest commit? If it looks good, maybe you can give me an approval 😄 
https://github.com/brave/brave-core/pull/7494/commits/8c1efaf717aa7c873a7e7ea71353b00e7d721f4a

File: components/brave_extension/extension/brave_extension/content_cosmetic.ts [hunk 3]
```diff
@@ -553,7 +553,7 @@ const scheduleQueuePump = (hide1pContent: boolean, generichide: boolean) => {
   }, { timeout: maxTimeMSBeforeStart })
 }
 
-const vettedSearchEngines = ['duckduckgo', 'qwant', 'bing', 'startpage', 'yahoo', 'onesearch', 'google', 'yandex']
+const vettedSearchEngines = ['duckduckgo', 'qwant', 'bing', 'startpage', 'yahoo', 'onesearch', 'google', 'yandex', 'ecosia']
```
Comment (pes10k) [inline components/brave_extension/extension/brave_extension/content_cosmetic.ts:556]: I think yahoo should be removed from here, right?

@SergeyZhukovsky , just a heads up to make sure the Android work you're doing stays in sync with the above changes

**Comment by pes10k** [comment]
@bsclifton just one small change requested, but otherwise the parts im familiar with look good to me!

File: components/brave_extension/extension/brave_extension/content_cosmetic.ts (hunk 3 above)
Comment (SergeyZhukovsky) [reply components/brave_extension/extension/brave_extension/content_cosmetic.ts:556]: sounds good, I added ecosia as well

**Comment by bsclifton** [comment]
@pes10k updated 😄 Please see https://github.com/brave/brave-core/pull/7494/commits/2020a478f8bd7a215b31a21829827ea08b401c39

**Comment by pes10k** [comment]
@bsclifton looks great!

**Comment by bsclifton** [comment]
CI looks good and we have approvals 😄 We might have a follow up to this - but will create a separate issue/PR for that

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x64` using the following build:


#### **`Case 1`** - Verify changes in the `Default search engines` list

**`USA`** | **`USA`** | **`USA`** | **`USA`**
-----------|-----------|-----------|-----------
SEUSA1 | SEUSA2 | SEUSA3 | SEUSA4

**`UK`** | **`UK`** | **`UK`** | **`UK`**
---------|---------|---------|---------
image | image | image | image

**`France`** | **`France`** | **`France` (Default Quant)** | **`France`**
-------------|---------------|-------------------------------|--------------
image | image | image | image

**`Note:`** ensured that `Qwant` was still the default search engine in `France`

**`Germany`** | **`Germany`** | **`Germany` (Default DDG)** | **`Germany`**
----------------|-----------------|-----------------------------------|--------------
image | image | image | image

**`Note:`** ensured that `DDG` was still the default search engine in `Germany`

**`Netherlands`** | **`Netherlands`** | **`Netherlands`** | **`Netherlands`**
----------------|-----------------|---------------------------|--------------
image | image | image | image

**`Belgium`** | **`Belgium`** | **`Belgium`** | **`Belgium`**
----------------|-----------------|---------------------------|--------------
image | image | image | image


**`Switzerland`** | **`Switzerland`** | **`Switzerland`** | **`Switzerland`**
----------------|-----------------|---------------------------|--------------
image | image | image | image

**`Sweden`** | **`Sweden`** | **`Sweden`** | **`Sweden`**
----------------|-----------------|---------------------------|--------------
image | image | image | image

#### **`Case 2`** - Verify folks w/ Yahoo set don't lose their setting

**`Case 1 - Canada`**

* downloaded `1.20.65 CR: 88.0.4324.51` and set the default SE as `Startpage`
* upgraded to `1.20.73 CR: 88.0.4324.79` and ensured that `Startpage` was still set as the default SE
  * checked both `brave://settings/search` & `brave://settings/searchEngines`
* ensured that `Ecosia` was being listed under `brave://settings/search` & `brave://settings/searchEngines`
* ensured that `Yahoo` was removed from both `brave://settings/search` & `brave://settings/searchEngines`

**`Case 2 - USA`**

* downloaded `1.20.65 CR: 88.0.4324.51` and set the default SE as `Yahoo`
* upgraded to `1.20.73 CR: 88.0.4324.79` and ensured that `Yahoo` was still set as the default SE
  * checked both `brave://settings/search` & `brave://settings/searchEngines`
* ensured that `Ecosia` was being listed under `brave://settings/search` & `brave://settings/searchEngines`

**`Case 3 - France`**

* downloaded `1.20.65 CR: 88.0.4324.51` and set the default SE as `Qwant` (Default)
* upgraded to `1.20.73 CR: 88.0.4324.79` and ensured that `Qwant` was still set as the default SE
  * checked both `brave://settings/search` & `brave://settings/searchEngines`
* ensured that `Ecosia` was being listed under `brave://settings/search` & `brave://settings/searchEngines`
* * ensured that `Yahoo` was removed from both `brave://settings/search` & `brave://settings/searchEngines`

#### **`Case 3`** - Verify the Ecosia referral IDs used are correct
1. Do a search (on Android/Desktop/using Ecosia
2. Verify `tt=` part of URL fragment matches referral code shown in https://github.com/brave/internal/wiki/Search-engine-integrations

Searched via the omnibox when `Ecosia` was set as the default search engine and ensured that `tt=e8eb07a6` as per https://github.com/brave/internal/wiki/Search-engine-integrations

* https://www.ecosia.org/search?tt=e8eb07a6&q={brave+software}
* https://www.ecosia.org/search?tt=e8eb07a6&q={BAT+token}

Ensured that directly going to `Ecosia` and initiating a search doesn't use the referral code as per the following:

* https://www.ecosia.org/search?q=Brave+software
* https://www.ecosia.org/search?q=BAT+token

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Samsung S10+` running `Android 10` using the following build:

#### **`Case 1`** - Verify changes in the `Default search engines` list

**`USA`** | **`USA`** | **`USA`** | **`USA`**
-----------|-----------|-----------|-----------
Screenshot_20210112-211049_Settings | Screenshot_20210112-211223_Brave - Nightly | Screenshot_20210112-211241_Brave - Nightly | Screenshot_20210112-211246_Brave - Nightly

**`UK`** | **`UK`** | **`UK`** | **`UK`**
---------|---------|---------|---------
Screenshot_20210112-212313_Settings | Screenshot_20210112-212457_Brave - Nightly | Screenshot_20210112-212507_Brave - Nightly | Screenshot_20210112-212512_Brave - Nightly

**`France`** | **`France` (Default Quant)** | **`France` (Default Quant)** | **`France`**
-------------|---------------|-------------------------------|--------------
Screenshot_20210112-213240_Settings | Screenshot_20210112-213149_Brave - Nightly | Screenshot_20210112-213202_Brave - Nightly | Screenshot_20210112-213207_Brave - Nightly

**`Note:`** ensured that `Qwant` was still the default search engine in `France`

**`Belgium`** | **`Belgium`** | **`Belgium`** | **`Belgium`**
----------------|-----------------|---------------------------|--------------
Screenshot_20210112-215904_Settings | Screenshot_20210112-215033_Brave - Nightly | Screenshot_20210112-215051_Brave - Nightly | Screenshot_20210112-215046_Brave - Nightly

**`Switzerland`** | **`Switzerland`** | **`Switzerland`** | **`Switzerland`**
----------------|-----------------|---------------------------|--------------
Screenshot_20210112-220213_Settings | Screenshot_20210112-220259_Brave - Nightly | Screenshot_20210112-220325_Brave - Nightly | Screenshot_20210112-220350_Brave - Nightly

**`Sweden`** | **`Sweden`** | **`Sweden`** | **`Sweden`**
----------------|-----------------|---------------------------|--------------
Screenshot_20210112-220927_Settings | Screenshot_20210112-221106_Brave - Nightly | Screenshot_20210112-221116_Brave - Nightly | Screenshot_20210112-221121_Brave - Nightly

#### **`Case 2`** - Verify folks w/ Yahoo set don't lose their setting <--- FAILED

**`Case 1 - Canada`**

* downloaded `1.20.65 CR: 88.0.4324.51` and set the following default SE's
  * `Standard Tab` --> `Startpage`
  * `Incognito Tab` --> `Bing`
* upgraded to `1.20.74 CR: 88.0.4324.79` and ensured that both `Startpage` & `Bing` were still set as the default SE
* ensured that `Ecosia` wasn't being listed under both `Standard Tab` & `Incognito Tab`
* ensured that `Yahoo` was removed from both  `Standard Tab` & `Incognito Tab`


* downloaded `1.20.65 CR: 88.0.4324.51` and set the following default SE's
  * `Standard Tab` --> `Startpage`
  * `Incognito Tab` --> `Bing`
* upgraded to `1.20.74 CR: 88.0.4324.79` and ensured that both `Startpage` & `Bing` were still set as the default SE
* ensured that `Ecosia` was being listed under both `Standard Tab` & `Incognito Tab`
* ensured that `Yahoo` was removed from both  `Standard Tab` & `Incognito Tab`

**Comment by kjozwiak** [comment]
@bsclifton quick update re: Germany. The onboarding modal doesn't appear due to https://github.com/brave/brave-browser/issues/11079. Believe a decision was made to leave it in that state as it would be difficult to change due to DDG being default in Germany. CCing @aekeus.

**Pull request merged**